test/test_closed_streams.py

```diff
@@ -48,24 +48,24 @@ def test_can_receive_multiple_rst_stream_frames(self, frame_factory):
 
         assert isinstance(event, h2.events.StreamReset)
 
-    def test_closed_stream_resets_further_frames(self, frame_factory):
+    def test_resett_closed_stream_resets_further_frames(self, frame_factory):
         """
-        A stream that is closed can receive further frames: it simply sends
-        RST_STREAM for it.
+        A stream that is closed by RST_STREAM can receive further frames:
+        it simply sends RST_STREAM for it.
         """
         c = h2.connection.H2Connection(client_side=False)
         c.receive_data(frame_factory.preamble())
         c.initiate_connection()
 
-        f = frame_factory.build_headers_frame(
+        header_frame = frame_factory.build_headers_frame(
             self.example_request_headers, flags=['END_STREAM']
         )
-        c.receive_data(f.serialize())
-        c.send_headers(
-            stream_id=1,
-            headers=self.example_response_headers,
-            end_stream=True
```

Comment:
This test is moved to class ```TestStreamClosedByRstStream``` at https://github.com/chhsiao90/hyper-h2/blob/handle-frames-on-closed-stream/test/test_closed_streams.py#L276-L323

Because currently we had closed stream that was closed by RST_STREAM or END_STREAM. So I separate them into different class.

And sorry for this massive PR that bring inconvenience to you :-(